src/app/auth/login.ts

```diff
@@ -113,6 +135,136 @@ export const loginWithPhoneCookie = async ({
   return kratosResult
 }
 
+export const loginUpgradeWithPhone = async ({
+  phone,
+  code,
+  ip,
+  account,
+}: {
+  phone: PhoneNumber
+  code: PhoneCode
+  ip: IpAddress
+  account: Account
+}): Promise<SessionToken | ApplicationError> => {
+  {
+    const limitOk = await checkFailedLoginAttemptPerIpLimits(ip)
+    if (limitOk instanceof Error) return limitOk
+  }
+
+  {
+    const limitOk = await checkFailedLoginAttemptPerPhoneLimits(phone)
+    if (limitOk instanceof Error) return limitOk
+  }
+
+  const validCode = await isCodeValid({ phone, code })
+  if (validCode instanceof Error) return validCode
+
+  await rewardFailedLoginAttemptPerIpLimits(ip)
+  await rewardFailedLoginAttemptPerPhoneLimits(phone)
+
+  // Scenario 1) Phone account already exists (no txn with device acct) - User logs in with device jwt then he
+  //             wants to upgrade to a phone account that already exists. there are no txns on the device account
+  //             Tell the user to logout and log back in with the phone account
```

Comment:
shouldn't we auto upgrade to the phone account here? seems awkward to ask the user log out if there is no tx.

I guess it requires whether the existing device has a tx. but that would be a common scenario and the ux would be a lot better.